### PR TITLE
MINOR: Update collaborators list

### PR DESCRIPTION
**Pull request by jlprat**
I did the following:

> git shortlog --email --numbered --summary --since=2023-07-24 > /tmp/contributors.txt

Then I excluded all committers from the list.

Cut the list to the top 10 and for each member I found their GitHub user name. List is in order of commits.

Note I used the `--email` flag as stated in https://issues.apache.org/jira/browse/KAFKA-14995

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by OmniaGM** [comment]
Nice PR one problem with the used command is that any one that has same email but different name attached to the commit don't count as one person. 

I tried this which basically drop the name and use emails only from the output of `git shortlog --email --numbered --summary --since=2023-07-24 `

```
git shortlog --email --numbered --summary --since=2023-07-24 | awk '
{
  # Extract email (the part between < and >)
  match($0, /<[^>]*>/);
  email = substr($0, RSTART+1, RLENGTH-2);

  # Add to the associative array
  commits[email] += $1;
}

END {
  # Print the result
  for (email in commits) {
    print commits[email], email;
  }
}
' | sort -nr
```

and got slightly different order.

Note my bash skills isn't the best :D

**Comment by jlprat** [comment]
I can give it a try and hope we don't have two people with the same name 😃

**Comment by chia7712** [comment]
maybe we should ping those active contributors to take a look at this PR. Otherwise, they could never know that permission.

**Comment by OmniaGM** [comment]
> I can give it a try and hope we don't have two people with the same name 

My script do count based on email only not names, so it should be safer and bit accurate. 
For example I tried the original command you shared and definitely my email is shown twice 
```
    27 Omnia Ibrahim <o.g.h.ibrahim@gmail.com>
    4	 Omnia G.H Ibrahim <o.g.h.ibrahim@gmail.com>
```

This == 31 which should be right after `34 Nikolay <nizhikov@apache.org>` after removing committers. 
Or between   `33 Kuan-Po (Cooper) Tseng <brandboat@gmail.com>` and  `29 Andrew Schofield <aschofield@confluent.io>` before removing committers. 

Same with others
```
    33	Colin Patrick McCabe <cmccabe@apache.org>
    11 	Colin P. McCabe <cmccabe@apache.org>
    
     33	Kuan-Po (Cooper) Tseng <brandboat@gmail.com>
     6	Kuan-Po Tseng <brandboat@gmail.com>
```

**Comment by OmniaGM** [comment]
I think the tricky part would be same person two emails :D but I can't find such one in the original `git shortlog --email --numbered --summary --since=2023-07-24` command yet. Hopefully we don't have this case

**Comment by chia7712** [comment]
> Same with others

another case:
```
20 kitingiao@gmail.com
12 51072200+frankvicky@users.noreply.github.com
```
Those active contributors have many alias :)

**Comment by chia7712** [comment]
How about using Github contributors page? 

https://github.com/apache/kafka/graphs/contributors?from=2023-07-24&to=2024-07-24&type=c

**Comment by frankvicky** [comment]
```
20 kitingiao@gmail.com
12 51072200+frankvicky@users.noreply.github.com 
```
Yes this is me.
This happened because I didn't know that GitHub has private email protection before... 😥

**Comment by OmniaGM** [comment]
> > Same with others
> 
> another case:
> 
> ```
> 20 kitingiao@gmail.com
> 12 51072200+frankvicky@users.noreply.github.com
> ```
> 
> Those active contributors have many alias :)

yup this is the tricky one I mentioned above same person different emails :D

**Comment by chia7712** [comment]
> How about using Github contributors page?

1. [FrankYang0529](https://github.com/FrankYang0529)
3. [kamalcph](https://github.com/kamalcph)
4. [apoorvmittal10](https://github.com/apoorvmittal10)
5. [lianetm](https://github.com/lianetm)
6. [brandboat](https://github.com/brandboat)
7. [kirktrue](https://github.com/kirktrue)
8. [nizhikov](https://github.com/nizhikov)
10. [OmniaGM](https://github.com/OmniaGM)
11. [dongnuo123](https://github.com/dongnuo123)
12. [frankvicky](https://github.com/frankvicky)

**Comment by OmniaGM** [comment]
> How about using Github contributors page?
> 
> https://github.com/apache/kafka/graphs/contributors?from=2023-07-24&to=2024-07-24&type=c

I think this is the most accurate one at least it count for these GitHub email and name issue. And for cases where someone is mixing different accounts then sadly this case can't be counted as one person.

**Comment by jlprat** [comment]
> maybe we should ping those active contributors to take a look at this PR. Otherwise, they could never know that permission.

Yes @chia7712 , I was planning on doing this once we had the changes merged to make sure the pinged people were the ones that made it to the list for sure.

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by jlprat** [comment]
I updated [KAFKA-14995](https://issues.apache.org/jira/browse/KAFKA-14995) to reflect this way to obtain the contributors that seems more fair than the one with emails or names.

**Comment by jlprat** [comment]
I'll merge this PR, as the pipelines don't check this file and we don't need to waste CPU cycles on this.


Mentioning the newly added collaborators here as an FYI - @FrankYang0529 @apoorvmittal10 @lianetm @brandboat @kirktrue @nizhikov @OmniaGM @dongnuo123 @frankvicky . Welcome all! Let us know if you have any questions!

You can check out the Collaborators section on https://kafka.apache.org/contributing, or this [dev mailing list thread](https://lists.apache.org/thread/93lb6jhkjkmb9op9629xt6c6olwym28c) for more details.

**Pull request merged**
